### ISSUE TITLE
Follow Users: Algorithm update for suggested follows

### ISSUE THREAD

**Issue by mirie** [forem/forem#18969]
## User Story (Who / What / Why)

As a user, I’d like to be able to browse a list of “suggested members to follow” in order to continue discovering quality users to engage with.

As an admin, I’d like to create Billboards that prompt members to consider browsing for suggested follows. This functionality gives me a destination to point people to — allowing flexibility in Billboard creation.

## Definition of Done

This task can be approached from a bottom-up perspective which focuses on the quality of the algorithm or top-down which first establishes the UX and works with existing algorithm.

*Note: Engineering advocated for adjusting the algorithm _before_ tackling #18873 in order to maximize the impact of the UI change and rollout and increase engagement. 

- [ ]  Smarter algorithm for logged-In users, showing authors that…
    - They have read recently
    - They don’t already follow
    - Sorted by descending follow count
- [ ] Establish `field_test` to determine bar for progress and make data-driven improvements over time.

This will be an opportunity to invest more in the “suggestion algorithm”, which can make onboarding smarter, as well as act as a proper “directory” for smaller communities (including stuff like [[forem.team](http://forem.team/)](http://forem.team) for example).

This could also inform functionality we could bake back into `/tags` which is currently *only* a global directory, but could also have a tab for “suggested”.

[Figma File with explorations](https://www.figma.com/file/4XToRbTBw5NyxYDxqXm4Ez/Follow-User-Experience?node-id=1209%3A16&t=0VoC11jmBNQR8DD7-4)

### Technical considerations

- As noted already, this is a good chance to consider what is re-usable here. We want to take this as a chance to invest more deeply in key components and algorithms.
- With the `field_test` we can establish a new test while re-using success criteria related to engagement and retention, for minimal additional code disruptions.

**Comment by mirie** [comment]
@pkfrank / @anujbhavsar96:

Some questions/feedback from the team when they were going about estimations:

> I think we need to really think through and investigate what data we can make the algorithm smarter on. Could we possibly get some friend/follow data from social logins? What are the possibilities based on the data we have access to? Perhaps this issue would benefit from a research task first.

> The implementation behind our feed algorithm was complex and took a lot of testing. What is our expectation on how well this suggestion algorithm works? how easy will future changes be applied to it?

@fdocr you noted that setting up the field test might be tricky / complicated -- could you expand more on this concern? @maestromac do you share similar concerns? I wasn't sure if your testing concerns were about just the tests to make sure the functionality was working OR if they also included the success calculations too

**Comment by maestromac** [comment]
@mirie yeah I share those concerns. Regarding testing, I meant the usage analytics/result after the rollout.

**Comment by fdocr** [comment]
> @fdocr you noted that setting up the field test might be tricky / complicated -- could you expand more on this concern?

My worry was rooted more in the implementation details of creating a `field_test` and along the lines of the feed algorithm complexity question/feedback mentioned too.

Depends a bit on whether we're measuring the _existing algorithm_ or only the new one. The existing one I believe would be [Users::SuggestForSidebar](https://github.com/forem/forem/blob/main/app/services/users/suggest_for_sidebar.rb). There are other user suggestion mechanisms in the codebase AFAICT but this is the one I see as best candidate for improving + reusing compared to others like recent users (I hope there's not other ones that I missed aside from manual selection).

Perhaps I'm not seeing the picture clearly of where the `field_test` would be "abstraction-wise". We have an [AbExperiment](https://github.com/forem/forem/blob/c6227fdadb9a187043a2cc4e1059bdda2b895676/app/models/ab_experiment.rb) class but it feels a little coupled to the main feed experiments. We could go with raw `field_test` calls (not using the [GoalConversionHandler](https://github.com/forem/forem/blob/c6227fdadb9a187043a2cc4e1059bdda2b895676/app/models/ab_experiment/goal_conversion_handler.rb)) where we need it, which would be clicks attributed to the suggestion variant (or something else?). On another hand we could embed this experiment only on the new page (to be implemented in #18873), so that measurement would only be part of that part 2 of the issue (until it exists).

All of the above to say that it would be great to hear other's thoughts and agree on a couple things:
1. Which is the _current algorithm_ that we're going to use as baseline or if the algorithm will be "written from scratch"
   - My vote is reuse [Users::SuggestForSidebar](https://github.com/forem/forem/blob/main/app/services/users/suggest_for_sidebar.rb) and improve it. Strong opinion weakly held in case others want were to argue otherwise 🙂 
2. Where are we placing the `field_test`
   - i.e. Site wide wherever the algorithm serves suggestions like onboarding/Members page/sidebar/etc (for max reach possible) or only on Member Page (easier to implement)
   - Since we're doing "bottom up" approach we could do site wide (might require some FE/BE work to put together clicks as conversions) and reuse one good algorithm across the site as much as possible instead of having logic spread out that does similar things
   - If time is a constraint then the easier approach of Member Page only would truly be less tricky and it sounds good to me too

I stand to be corrected on anything of the above since this is a cool and interesting area (abtests) but somewhat advanced in terms of abstractions IMO 😄
